src/core/fstream.cc

```diff
@@ -292,16 +292,16 @@ class file_data_source_impl : public data_source_impl {
             auto len = end - start;
             auto actual_size = std::min(end - _pos, _remain);
             _read_buffers.emplace_back(_pos, actual_size, futurize_invoke([&] {
-                    return _file.dma_read_bulk<char>(start, len, _options.io_priority_class, &_intent);
+                    return _file.dma_read_bulk_impl(start, len, _options.io_priority_class, &_intent);
             }).then_wrapped(
-                    [this, start, pos = _pos, remain = _remain] (future<temporary_buffer<char>> ret) {
+                    [this, start, pos = _pos, remain = _remain] (future<temporary_buffer<uint8_t>> ret) {
```

Comment:
> Are all these char/uint8_t changes and casts necessary? 

Yes, because compiler fails to auto-convert buffers of <char> and <uint8_t> to each other

> I guess dma_read_bulk_impl uses uint8_t but for some reason we really want this function to work with char? Do we have to?

That's a good question and I've no answer for it. The guess is that `dma_read_bulk_impl()` is called from `template <CharType> dma_read_bulk()` so we have no guarantee that the _impl thing will match any `CharType` from the API method. Thus it was selected to be 'generic enough' (88b4fab1dd91e52701d65106a99985949100be95)